src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -0,0 +1,999 @@
+				// TODO: routing control for screen destination
+				break;
+			case SUPER_AUX_DESTINATION:
+				controlAssignSourceToSuperAuxDest(value, propertyValue);
+				break;
+			default:
+				if (logger.isWarnEnabled()) {
+					logger.warn(String.format("Operation %s with value %s is not supported.", property, value));
+				}
+				throw new IllegalArgumentException(String.format("Operation %s with value %s is not supported.", property, value));
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 * Control properties
+	 *
+	 * @param list list of control properties that will be controlled
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> list) {
+		if (CollectionUtils.isEmpty(list)) {
+			throw new IllegalArgumentException("Controllable properties cannot be null or empty");
+		}
+		for (ControllableProperty controllableProperty : list) {
+			controlProperty(controllableProperty);
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 * This method is called by Symphony to get the list of statistics to be displayed
+	 *
+	 * @return List<Statistics> This returns the list of statistics
+	 * @throws Exception Throw exception when failed to get info from device
+	 */
+	@Override
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		if (logger.isDebugEnabled()) {
+			logger.debug(String.format("Perform doPost() at host %s with port %s", this.host, this.getPort()));
+		}
+		Map<String, String> statistics = new HashMap<>();
+		List<AdvancedControllableProperty> controls = new ArrayList<>();
+		if (localExtendedStatistics == null) {
+			localExtendedStatistics = new ExtendedStatistics();
+		} else {
+			statistics = localExtendedStatistics.getStatistics();
+		}
+		initializeData(statistics, controls);
+		localExtendedStatistics.setStatistics(statistics);
+		localExtendedStatistics.setControllableProperties(controls);
+		return Collections.singletonList(localExtendedStatistics);
+	}
+
+	/**
+	 * Call post request on the device to get RpcResponse DTO
+	 *
+	 * @param method String name of the method
+	 * @param param Map of params
+	 * @return JsonNode returns the JsonNode of given object
+	 * @throws Exception Throw exception when failed to call post request, get data from device
+	 */
+	protected JsonNode getByMethod(String method, Map<Object, Object> param) throws Exception {
+		RpcResponse rpcResponse;
+		try {
+			rpcResponse = this.doPost(BarcoE2Constant.DOUBLE_QUOTES, rpcRequestBody(method, param), RpcResponse.class);
+			if (rpcResponse == null || rpcResponse.getSuccessCode() != 0) {
+				throw new ResourceNotReachableException("doPost success but failed to get data from the device");
+			}
+			JsonNode response = rpcResponse.getResponse();
+			if (response != null && !NullNode.instance.equals(response) && !response.isEmpty()) {
+				return response;
+			}
+			return objectMapper.valueToTree(rpcResponse);
+		} catch (Exception e) {
+			logger.error(String.format("Failed to doPost for method %s", method));
```

Comment:
is it necessary to have an extra log entry before the rethrow? Maybe catch block is not needed? 
Also the ResourceNotReachableException from above will be caught by it

Reply:
Updated.